### PR TITLE
added ability to filter based on color and size

### PR DESCRIPTION
**Pull request by AdamWinek**
Added new card section with filters allowing users to filter by color and size.

### CODE REVIEW

File: controllers/items.js [hunk 1]
```diff
@@ -15,7 +16,68 @@ router.get('/', [userIsBasicUser], async function (req, res, next) {
...
+    sizes = Object.values(size)
+    isALlUndefined = true
+    sizes.forEach(element => {
+        if (element != undefined) {
+            isALlUndefined = false
+        }
+    });
+    return isALlUndefined
+}
+
```
Comment (jackgoldsmith4) [inline controllers/items.js]: whitespace

File: controllers/items.js [hunk 2]
```diff
@@ -15,7 +16,68 @@ router.get('/', [userIsBasicUser], async function (req, res, next) {
...
+    let shirtSize = req.query.shirtSize
+    //used for pants
+    let waistSize = req.query.waistSize
+    let pantsLength = req.query.pantsLength
+    //used for shoes
+    let shoeSize = req.query.shoeSize
+    let size = { shoeSize: shoeSize, chestSize: chestSize, sleeveSize: sleeveSize, shirtSize: shirtSize, waistSize: waistSize, pantsLength: pantsLength }
+    let response = {};
+    try {
+        console.log(Object.values(size))
```
Comment (jackgoldsmith4) [inline controllers/items.js]: console.log

File: controllers/items.js [hunk 3]
```diff
@@ -15,7 +16,68 @@ router.get('/', [userIsBasicUser], async function (req, res, next) {
...
+            response.items = await itemService.getItemCategorySizeColor(req.params.type, req.params.gender, size, Object.keys(req.query.color))
+            console.log(response.items)
+        } else if (req.query.color != undefined) {
+            response.items = await itemService.getItemsByCategoryAndColor(req.params.gender, req.params.type, Object.keys(req.query.color))
+
+
+        } else if (!hasNoSize(size)) {
+            response.items = await itemService.getItemCategorySize(req.params.type, req.params.gender, size)
+        }
+
```
Comment (jackgoldsmith4) [inline controllers/items.js]: whitespace within this function

File: db/sequelize.js [hunk 4]
```diff
@@ -37,6 +37,7 @@ if (process.env.POSTGRESQL_SERVICE_PORT) {
     options.port = process.env.DATABASE_PORT;
 }
 
+
```
Comment (jackgoldsmith4) [inline db/sequelize.js:40]: delete

File: services/item-service.js [hunk 5]
```diff
@@ -14,6 +14,9 @@ const { v4: uuidv4 } = require("uuid"),
     InternalErrorException = require("../exceptions/internal-error-exception"),
     exceptionHandler = require("../exceptions/exception-handler"),
     csvParser = require("csv-parse");
+const { suits } = require("../db/sequelize");
+var Op = Sequelize.Op;
+
```
Comment (jackgoldsmith4) [inline services/item-service.js]: whitespace

File: services/item-service.js [hunk 6]
```diff
@@ -125,13 +128,156 @@ exports.getPants = async function (itemId) {
...
+                model: Suits, as: "suits"
+            }]
+        });
+        console.log(items)
+        return items;
+    } catch (e) {
+        throw new InternalErrorException("A problem occurred when retrieving items", e);
+    }
+}
+
```
Comment (jackgoldsmith4) [inline services/item-service.js]: whitespace

File: services/item-service.js [hunk 7]
```diff
@@ -125,13 +128,156 @@ exports.getPants = async function (itemId) {
...
+            case "suits":
+                items = await Item.findAll({
+                    where: { gender: gender, type: type },
+                    include: [{
+                        model: Suits, as: "suits"
+                    }]
+                });
+                break;
+
```
Comment (jackgoldsmith4) [inline services/item-service.js]: whitespace between each case

File: services/item-service.js [hunk 8]
```diff
@@ -125,13 +128,156 @@ exports.getPants = async function (itemId) {
...
+                    }]
+                });
+                break
+        }
+        return items;
+    } catch (e) {
+        throw new InternalErrorException("A problem occurred when retrieving items", e);
+    }
+}
+
```
Comment (jackgoldsmith4) [inline services/item-service.js]: lots of space between functions...this is just my preference but if you want this to stay then idc

File: services/item-service.js [hunk 9]
```diff
@@ -125,13 +128,156 @@ exports.getPants = async function (itemId) {
...
+                        gender: gender, type: type, color: {
+                            [Op.in]: colors
+                        }
+                    },
+                    include: [{
+                        model: Shoes, as: "shoes"
+                    }]
+                });
+                break;
+
```
Comment (jackgoldsmith4) [inline services/item-service.js:187]: whitespace between cases

File: services/item-service.js [hunk 10]
```diff
@@ -125,13 +128,156 @@ exports.getPants = async function (itemId) {
...
+                });
+                break
+        }
         return items;
     } catch (e) {
         throw new InternalErrorException("A problem occurred when retrieving items", e);
     }
 }
 
+
```
Comment (jackgoldsmith4) [inline services/item-service.js]: delete

File: services/item-service.js [hunk 11]
```diff
@@ -245,6 +395,225 @@ exports.getItemAndSize = async function (type, gender, color, brand, size) {
...
+                break;
+            case "shirts":
+                item = await Item.findAll({
+                    where: { type: type, gender: gender }, include: [{
+                        model: Shirts,
+                        where: { size: size.shirtSize },
+                        as: "shirts"
+                    }]
+                });
+
```
Comment (jackgoldsmith4) [inline services/item-service.js:376]: delete line

File: services/item-service.js [hunk 12]
```diff
@@ -245,6 +395,225 @@ exports.getItemAndSize = async function (type, gender, color, brand, size) {
...
+ * @param {enum} type
+ * @param {enum} gender
+ * @param {object} size
+ *  if object is of type suits: size has chestSize and sleeveSize attributes
+ * if object is of type shirt: size has shirtSize attribute 
+ * if object is of type pants: size has waistSize and pantsLength attributes
+ * if object is of type shoes: size has an shoeSize attribute
+ */
+exports.getItemCategorySize = async function (type, gender, size) {
+
```
Comment (jackgoldsmith4) [inline services/item-service.js:354]: delete empty line

File: services/item-service.js [hunk 13]
```diff
@@ -245,6 +395,225 @@ exports.getItemAndSize = async function (type, gender, color, brand, size) {
...
+                break;
+        }
+
+        return item
+    } catch (e) {
+        throw e;
+    }
+}
+
+
```
Comment (jackgoldsmith4) [inline services/item-service.js]: whitespace

File: services/item-service.js [hunk 14]
```diff
@@ -245,6 +395,225 @@ exports.getItemAndSize = async function (type, gender, color, brand, size) {
...
+ * @param {enum} type
+ * @param {enum} gender
+ * @param {object} size
+ *  if object is of type suits: size has chestSize and sleeveSize attributes
+ * if object is of type shirt: size has shirtSize attribute 
+ * if object is of type pants: size has waistSize and pantsLength attributes
+ * if object is of type shoes: size has an shoeSize attribute
+ */
+exports.getItemCategorySize = async function (type, gender, size) {
+
```
Comment (jackgoldsmith4) [inline services/item-service.js:416]: delete empty line

File: services/item-service.js [hunk 15]
```diff
@@ -245,6 +395,225 @@ exports.getItemAndSize = async function (type, gender, color, brand, size) {
...
+                break;
+        }
+
+        return item
+    } catch (e) {
+        throw e;
+    }
+}
+
+
```
Comment (jackgoldsmith4) [inline services/item-service.js]: whitespace

File: services/item-service.js [hunk 16]
```diff
@@ -245,6 +395,225 @@ exports.getItemAndSize = async function (type, gender, color, brand, size) {
...
```
Comment (jackgoldsmith4) [inline services/item-service.js]: empty lines

File: views/admin/entry-search.ejs [hunk 17]
```diff
@@ -49,19 +75,19 @@
                 <button type="button" class="btn btn-danger" title="Remove" aria-label="Remove" data-toggle="modal"
                   data-target="#removeModal" data-id="<%= item.id %>" data-name="<%= item.name %>"
                   data-barcode="<%= item.barcode %>" data-count="<%= item.count %>"
-                  data-description="<%= item.description %>"><i class="fas fa-minus"></i></button>
+                  data-description="<%= item.description %>"><i class="fas fa-minus"></i></button> -->
                 <!-- edit button -->
-                <button type="button" class="btn btn-secondary" title="Edit" aria-label="Edit" data-toggle="modal"
+                <!-- <button type="button" class="btn btn-secondary" title="Edit" aria-label="Edit" data-toggle="modal"
```
Comment (jackgoldsmith4) [inline views/admin/entry-search.ejs]: are these comments going to stay?

File: views/components/header.ejs [hunk 18]
```diff
@@ -43,6 +68,11 @@
       <li class="nav-item rounded float-right">
         <a class="nav-link rounded" href="/cart"><i class="fas fa-shopping-cart"></i> Cart</a>
       </li>
+      
+
+
```
Comment (jackgoldsmith4) [inline views/components/header.ejs]: whitespace

File: views/user/view-items.ejs [hunk 19]
```diff
@@ -13,41 +13,193 @@
...
-				</tbody>
-			</table>
+								<% } %>
+							<% }); %>
+							<% } %>
+						<!-- </tbody>
+					</table> -->
+				</div>
+
+
```
Comment (jackgoldsmith4) [inline views/user/view-items.ejs]: whitespace

**Review by jackgoldsmith4 — CHANGES_REQUESTED** [review]

File: db/sequelize.js (hunk 4 above)
Comment (AdamWinek) [reply db/sequelize.js:40]: This is needed so we can test and also use the db in prod

**Comment by AdamWinek** [comment]
Removed whitespace and fixed error with items test

**Review by jackgoldsmith4 — APPROVED** [review]

**Pull request merged**